backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -0,0 +1,207 @@
+package org.ovirt.engine.core.bll.scheduling.utils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.common.businessentities.CpuPinningPolicy;
+import org.ovirt.engine.core.common.businessentities.VDS;
+import org.ovirt.engine.core.common.businessentities.VM;
+import org.ovirt.engine.core.common.businessentities.VdsCpuUnit;
+import org.ovirt.engine.core.common.utils.CpuPinningHelper;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.vdsbroker.ResourceManager;
+
+public class VdsCpuUnitPinningHelper {
+    @Inject
+    private ResourceManager resourceManager;
+
+    /**
+     * This function will tell if the host is capable to run a given dedicated CPU policy VM.
+     *
+     * The function first will pin the CPU pending resources to the host, then:
+     * 1. If the VM number of vCPUs is smaller or equal to the pCPU exists on a single socket:
+     *    Pass on each socket and check if there are enough free pCPU to use.
+     *    If there are enough pCPUs - return true, otherwise false.
+     * 2. If the VM number of vCPUs is bigger to the pCPU exists on a single socket:
+     *    - We must use multiple sockets.
+     *    Pass on the host and check if there are enough free pCPU to use (regardless being in multiple sockets).
+     *    If there are enough pCPUs - return true, otherwise false.
```

Comment:
so you were looking for a quick filtering on the host, right?
I'm not sure about the logic though.
as for the first part, if I take a VM with 2:1:1 (sockets:cores:threads), meaning 2 sockets and overall 2 vCPUs - will hosts with 4:1:1 be filtered out?
as for the second part, don't you want to filter out hosts that we'll need to use more pSockets than the vSockets on?

Reply:
I do not understand the logic - if the number of vCPUs < number of pCPUs in a socket, the CPUs cannot be split between different sockets. If the number of vCPUs > number of pCPUs, it can be split? Why is that?

Reply:
> I do not understand the logic - if the number of vCPUs < number of pCPUs in a socket, the CPUs cannot be split between different sockets. If the number of vCPUs > number of pCPUs, it can be split? Why is that?

If we have an hardware that can fit all the vCPUs into a single socket this is what we want, nothing else. Otherwise, we cause more harm (performance) instead of what we really want. However, if we can't fit it to single pSocket, we must split it - or we can't run such VM at all. Using multiple pSockets is fine, but we don't want that in case it can fit into a single one.

- This was a bottom line of discussions about scheduling dedicated vm.

Reply:
I discussed about it offline with Arik.
We will keep the current logic (trying to fit into one socket), if that won't succeed, we will look into fitting it to the virtual sockets.

Reply:
Going to be on a follow-up PR.